src/components/show-hide-when/display-when.ts

```diff
@@ -1,46 +1,134 @@
-import { NgZone } from '@angular/core';
+import { Input, NgZone } from '@angular/core';
 
+import { Config } from '../../config/config';
 import { Platform } from '../../platform/platform';
 
+const QUERY: { [key: string]: string }  = {
+  xs: '(min-width: 0px)',
+  sm: '(min-width: 576px)',
+  md: '(min-width: 768px)',
+  lg: '(min-width: 992px)',
+  xl: '(min-width: 1200px)'
+};
 
 /**
  * @hidden
  */
 export class DisplayWhen {
+  private _resizeObs: any;
+  private _query: string;
+  private _rmListener: any;
+
   isMatch: boolean = false;
   conditions: string[];
-  resizeObs: any;
 
-  constructor(conditions: string, public _plt: Platform, public zone: NgZone) {
+  @Input()
+  set when(query: string) {
+    if (typeof query === 'boolean') {
+      this.isMatch = this._query = query;
+    } else {
+      query = this._query = query.toLowerCase();
+      this.checkConditions(query);
+    }
+  }
+
+  get when(): string {
+    return this._query;
+  }
+
+  constructor(conditions: string, public _config: Config, public _plt: Platform, public zone: NgZone) {
+    // this.checkConditions(conditions);
+  }
+
+  checkConditions(conditions: string): void {
     if (!conditions) return;
 
-    this.conditions = conditions.replace(/\s/g, '').split(',');
+    // Split the conditions on commas, unless the comma is inside of parantheses
+    // because this means it is inside of a media query
+    this.conditions = conditions.replace(/\s/g, '').split(/\,(?![^\(]*\))/g);
+    console.log('conditions', this.conditions);
 
     // check if its one of the matching platforms first
     // a platform does not change during the life of an app
     for (let i = 0; i < this.conditions.length; i++) {
-      if (this.conditions[i] && _plt.is(this.conditions[i])) {
-        this.isMatch = true;
-        return;
+      console.log('condition', this.conditions[i]);
+
+      // media: xs, media: (min-width: 44px), media: print and (min-width: 450px)
+      if (this.conditions[i] && this.conditions[i].indexOf('media') > -1) {
+        this.compareMedia(this.conditions[i]);
```

Comment:
add: continue;

if the condition matches there is no need to checking for mode